src/js/actions/tools.js

```diff
@@ -881,7 +887,7 @@ define(function (require, exports) {
      */
     var onReset = function () {
         descriptor.removeListener("toolModalStateChanged", _toolModalStateChangedHandler);
-
+        this.flux.store("document").removeListener("change", _documentChangeHandler);
```

Comment:
Need to listen to application store changes as well to account for changes to the active document.